### PR TITLE
[Navigation] Update context control background on small screens

### PR DESCRIPTION
**Pull request by mateus**
### WHY are these changes introduced?

Update that was overlooked during the first iteration.

### WHAT is this pull request doing?

Update context control background on small screens to match the TopBar

**Before**

Screenshot 2023-06-26 at 3 33 10 PM
text control background on mobile.

**After**

Screenshot 2023-06-26 at 3 32 46 PM

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by chloerice — APPROVED** [review]

File: .changeset/warm-balloons-impress.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Updates Navigation's context control background on small screens
```
Comment (kyledurand) [inline .changeset/warm-balloons-impress.md]: We don't need a changelog entry for se23 stuff

**Review by kyledurand — APPROVED** [review]

**Pull request merged**